src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -57,31 +55,51 @@ public sealed class ODataBatchReader : IODataBatchOperationListener
         /// <summary>
         /// Internal switch for whether we support reading Content-ID header appear in HTTP head instead of ChangeSet head.
         /// </summary>
-        private bool allowLegacyContentIdBehaviour;
+        private bool allowLegacyContentIdBehavior;
+
+        protected ODataInputContext InputContext
+        {
+            get { return this.inputContext; }
+        }
+
+        protected string ContentIdToAddOnNextRead
+        {
+            get { return this.contentIdToAddOnNextRead; }
+            set { this.contentIdToAddOnNextRead = value; }
+        }
+
+        protected OperationState ReaderOperationState
+        {
+            get { return this.operationState; }
+            set { this.operationState = value; }
+        }
+
+        protected bool AllowLegacyContentIdBehavior
+        {
+            get { return this.allowLegacyContentIdBehavior;}
+        }
 
         /// <summary>
         /// Constructor.
         /// </summary>
         /// <param name="inputContext">The input context to read the content from.</param>
         /// <param name="batchBoundary">The boundary string for the batch structure itself.</param>
-        /// <param name="batchEncoding">The encoding to use to read from the batch stream.</param>
         /// <param name="synchronous">true if the reader is created for synchronous operation; false for asynchronous.</param>
-        internal ODataBatchReader(ODataRawInputContext inputContext, string batchBoundary, Encoding batchEncoding, bool synchronous)
+        internal ODataBatchReader(ODataInputContext inputContext, bool synchronous)

```

Comment:
>internal [](start = 8, length = 8)

If ODataBatchReader is abstract, shouldn't this constructor be protected?  Shouldn't code always create one of the derived types?